lib/linter.js

```diff
@@ -307,6 +307,7 @@ class Linter {
 
     let hasBOM = options.source.charCodeAt(0) === 0xfeff;
     let source = hasBOM ? options.source.slice(1) : options.source;
+    source = source.replace(/<!doctype[^>]+>\n?/i, '');
```

Comment:
We need to keep track of the original doctypes, so that we can put it back for fixed templates

Reply:
Done. Please verify.